sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala

```diff
@@ -818,7 +818,8 @@ class Analyzer(
           failAnalysis(s"$expr has multiple Window Specifications ($distinctWindowSpec)." +
             s"Please file a bug report with this error message, stack trace, and the query.")
         } else {
-          distinctWindowSpec.head
+          val unbounded = SpecifiedWindowFrame(RowFrame, UnboundedPreceding, UnboundedFollowing)
```

Comment:
Let's add a comment at here to explain what we are doing. Seems, we create a dummy window frame, so we only use `PARTITION BY` clause and `ORDER BY` clause to do grouping.

Actually, can we just extract partitioning expressions and ordering expressions and explicitly use them as the grouping key? That will make the code easier to understand. Also, we probably want to remove the argument `windowSpec` from both logical and physical Window operator because we create `WindowFunctionFrame` based on the frame definition of every WindowExpression's frame definition.